### PR TITLE
Upgrade validator

### PR DESCRIPTION
**Pull request by GalaxySciTech**
# Proposed changes
This is for validator contract upgrade

## Types of changes

What types of changes does your code introduce to XDC network?
_Put an `✅` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Regular KTLO or any of the maintaince work. e.g code style
- [ ] CICD Improvement

## Impacted Components
Which part of the codebase this PR will touch base on,

_Put an `✅` in the boxes that apply_

- [ ] Consensus
- [ ] Account
- [ ] Network
- [ ] Geth
- [x] Smart Contract
- [ ] External components
- [ ] Not sure (Please specify below)

## Checklist
_Put an `✅` in the boxes once you have confirmed below actions (or provide reasons on not doing so) that_

- [ ] This PR has sufficient test coverage (unit/integration test) OR I have provided reason in the PR description for not having test coverage
- [ ] Provide an end-to-end test plan in the PR description on how to manually test it on the devnet/testnet.
- [ ] Tested the backwards compatibility.
- [ ] Tested with XDC nodes running this version co-exist with those running the previous version.
- [ ] Relevant documentation has been updated as part of this PR
- [x] N/A

### CODE REVIEW

File: contracts/validator/src/contracts/XDCValidator.sol [hunk 1]
```diff
@@ -306,114 +328,84 @@ contract XDCValidator {
...
-                    delete kycString[_invalidMasternode];
-                    delete ownerToCandidate[_invalidMasternode];
-                    delete invalidKYCCount[_invalidMasternode];
+            invalidOwner[_owner] = true;
+
+            (bool isOwnerNow, uint ownerIndex) = isOwner(_owner);
+            if (isOwnerNow) {
+                uint j = 0;
+                uint count = 0;
+                address[] memory allMasternodes = new address[](
```
Comment (wgr523) [inline contracts/validator/src/contracts/XDCValidator.sol]: suggest to rename allMasternodes to allInvalidMasternodes

File: contracts/validator/src/test/xdcValidator.js [hunk 2]
```diff
@@ -112,7 +112,6 @@ describe("XDCValidator", () => {
       expect(candidatesBefore).to.deep.eq(before);
       expect(candidates).to.deep.eq(deploy["candidates"]);
       expect(voters).to.include(master);
-      expect(ownerToCandidates).to.include(candidate);
```
Comment (wgr523) [inline contracts/validator/src/test/xdcValidator.js:115]: why remove this

Comment (GalaxySciTech) [reply contracts/validator/src/test/xdcValidator.js:115]: before this is bug , resign should delete all info , but old version didn't delete

Comment (wgr523) [reply contracts/validator/src/test/xdcValidator.js:115]: do we have "expect not include"?

Comment (GalaxySciTech) [reply contracts/validator/src/test/xdcValidator.js:115]: yes changed

**Review by wgr523 — APPROVED** [review]

**Pull request merged**